core_lib/src/graphics/bitmap/bitmapimage.cpp

```diff
@@ -876,7 +886,28 @@ void BitmapImage::floodFill(BitmapImage* targetImage,
         }
     }
 
-    targetImage->paste(replaceImage);
+    switch(fillMode)
+    {
+    default:
+    case 0:
+        targetImage->paste(replaceImage);
+        break;
+    case 1:
+        BitmapImage fullOpacity(*replaceImage);
+        if (qAlpha(newColor) == 0xFF)
+        {
+            targetImage->paste(replaceImage);
+        }
+        else
+        {
+            targetImage->paste(replaceImage, QPainter::CompositionMode_DestinationOut);
+            BitmapImage properColor(targetImage->mBounds, QColor::fromRgba(newColor));
+            properColor.paste(replaceImage, QPainter::CompositionMode_DestinationIn);
+            targetImage->paste(&properColor);
+        }
+        break;
+    }
+
```

Comment:
Same as above, i'd prefer if we moved this logic out of the floodfill method as it has nothing to do with the algorithm.

Reply:
Done